src/schemas/HederaUtils.ts

```diff
@@ -249,3 +249,51 @@ export function lookupNFTTransfer(transaction: Transaction, tokenId: string): Nf
     }
     return result
 }
+
+export function decodeRedirectForTokenInput(inputArgs: string): ethers.Result {
+    const tokenAddress = `0x${inputArgs.slice(2, 42)}`
+    const encodedFunctionSelector = `0x${inputArgs.slice(42)}`
+    return new ethers.Result(tokenAddress, encodedFunctionSelector)
+}
+
+export function resolveFunctionFragmentForHTSProxyContract(functionFragment: ethers.FunctionFragment, inputArgs: string): ethers.FunctionFragment {
+    let encodedFunction4BytesSignature = ""
+
+    try {
+        const inputResult = ethers.AbiCoder.defaultAbiCoder().decode(functionFragment.inputs, inputArgs)
+        encodedFunction4BytesSignature = inputResult[1].slice(0, 10)
+    } catch (failure) {
+        const f = failure as ethers.EthersError
+        if (f.code === "BUFFER_OVERRUN" || f.code === "INVALID_ARGUMENT") {
+            const inputResult = decodeRedirectForTokenInput(inputArgs)
+            encodedFunction4BytesSignature = inputResult[1].slice(0, 10)
+        }
+    }
+
```

Comment:
Why should we attempt calling `AbiCoder.decode()` in this context ?
We know that a special decoding rule is needed ; so let's do it straight.

```

export function resolveFunctionFragmentForHTSProxyContract(functionFragment: ethers.FunctionFragment, inputArgs: string): ethers.FunctionFragment {
    const inputResult = decodeRedirectForTokenInput(inputArgs)
    const encodedFunction4BytesSignature = inputResult[1].slice(0, 10)

    // @notice this is the list of supported method which is collected based on HIP-218 https://hips.hedera.com/hip/hip-218
    const ABI_FOR_SUPPORTED_METHODS = [
...

```